core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -465,6 +466,10 @@ private void runInternal() {
           LOG.error("Failed to close the request.", e);
         }
         replyError(error);
+        Throwable rootCause = error.getCause().getCause();
+        if (rootCause instanceof java.lang.Error) {
+          ProcessUtils.fatalError(LOG, rootCause, "Fatal Error caught while reading");
```

Comment:
what are the cases when we want the worker to just crash? `OutOfMemoryError`?

what if the worker isn't managed by Kubernetes?

Reply:
I think it's fine to let the worker suicide when there's a fatal error and let DevOps handle the failure. We should not allow false prosperity. The master can also monitor worker's state to decide what to do but since worker knows themselves best so I would prefer to let the worker decide first. Ideally, we should have both mechanisms (master&worker) to keep the cluster in a healthy state